lib/rust/ensogl/core/src/display/symbol/gpu/geometry/compound/sprite.rs

```diff
@@ -213,12 +214,13 @@ impl Deref for Sprite {
 #[derive(Clone, CloneRef, Debug)]
 #[allow(missing_docs)]
 pub struct SpriteSystem {
-    pub symbol: Symbol,
-    transform:  Buffer<Matrix4<f32>>,
-    uv:         Buffer<Vector2<f32>>,
-    size:       Buffer<Vector2<f32>>,
-    alignment:  Uniform<Vector2<f32>>,
-    stats:      Stats,
+    pub symbol:   Symbol,
+    transform:    Buffer<Matrix4<f32>>,
+    uv:           Buffer<Vector2<f32>>,
+    size:         Buffer<Vector2<f32>>,
+    alignment:    Uniform<Vector2<f32>>,
+    stats:        Stats,
+    registration: Rc<SymbolRegistration>,
 }
```

Comment:
1. Why this thing is done in `SpriteSystem`? If I understand corrctly, this is meant to manage symbol registry cleanup, right? If so, why is it not done in Symbol instead, which is a more low-level concept than SpriteSystem.
2. Alternatively, we can clean-up symbol registry during rendering - we are then iterating over all symbols in symbol registry. Such a solution would not require to have so complex circular dependencies. Does your solution has any advantages over this one?

Reply:
After giving `Shader` ownership some more thought, I have implemented (2).